src/assemblyItemsBrowser.js

```diff
@@ -192,13 +198,58 @@ module.exports = React.createClass({
 		}
 		
 		var currentAddress = this.state.currentAddress
-		if (props.vmTrace[vmTraceIndex].address) // there's a new context
-		{
-			var stack = this.state.callStack[vmTraceIndex]
-			currentAddress = stack[stack.length - 1]
-			this.resolveAddress(currentAddress, props)
-			Object.assign(stateChanges, { "currentAddress": currentAddress })
-		}
+        var addressIndex = this.shouldUpdateStateProperty("address", vmTraceIndex, previousIndex, props.vmTrace)
+        if (addressIndex > -1)
+        {
+            if (this.state.currentSelected === -1) // beginning of the trace
+            {
+                currentAddress = props.vmTrace[addressIndex].address
+                if (props.transaction.to === null) // contract creation
+                {
+                    this.cacheCode(currentAddress, props.transaction.input)
```

Comment:
The code is not stored there but we need to cache it (even if it comes from transaction.input). and associate it with a key that allow us to retrieve it later.
If you associate code with stack frame, there will be no caching.
For example we are landing on a new CALLCODE, there's a new code address, so we retrieve it.
Several steps after, we land on a different CALLCODE but the same address, we should be able to retrieve the code easily cause we've already fetched it. (and the common way will be to fetch it using the address). 
maybe the name of the function is not really accurate. it should be cacheExecutingCode.